src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -366,17 +365,17 @@ public ShellDecorator withShell(String shell) {
 
     @Override
     public String prepareIpcFilepath(Path rootPath) {
-        return NAMED_PIPE;
+        return NAMED_PIPE + rootPath.toAbsolutePath();
```

Comment:
This change is so that a unit test can for sure connect to the intended IPC server that has the mock setup. Without this change, unit test may connect to a different IPC server that is started by a different test, and for whatever reason, still running.

This change is likely the only one that is absolutely necessary to make the test to pass.

Reply:
Can this path end up being longer than the max pipe name length? We've faced similar issues with the socket file paths in UATs earlier. Could making the named pipe non static and initializing with a UUID appended to the static prefix also work instead?

Reply:
Good call. I found that the max length of a named pipe is 256. If the the prefix+rootPath is <= 256, I will use that string. Otherwise, I will use the prefix+UUID. The UUID needs to be static because there are a few places that need the same UUID.

Reply:
This should work, although sounds a bit unnecessary to have two code paths. I was suggesting we don't use the rootpath at all, instead could have a default constructor for WindowsPlatform, which would initialize a namedPipe member as `NAMED_PIPE + UUID.randomUUID().toString()` and whenever you have to get the file path you return that so it will be consistent everywhere, since a platform instance gets created only once, but that's to your judgement

Reply:
The problem I am trying to fix is more related to running tests: 
1. The name must be unique to one test only. Otherwise, the test could be connecting to an orphan IPC server.
2. The name must be consistent within the test.

Using UUID as a static suffix cannot solve problem 1. Using a new UUID every time cannot solve problem 2. Using the `rootPath` as the suffix is ideal that it solves both problems. If `rootPath` is too long, I am hoping it is outside the scope of running tests. And therefore static UUID would do.